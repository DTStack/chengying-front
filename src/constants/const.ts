// 部署状态
export const deployStatus = {
};

// 部署状态filter
export const deployStatusFilter = [
  {
    text: '部署成功',
    value: 'deployed',
  },
  {
    text: '部署失败',
    value: 'deploy fail',
  },
  {
    text: '部署中',
    value: 'deploying',
  },
  {
    text: '卸载失败',
    value: 'undeploy fail',
  },
  {
    text: '卸载中',
    value: 'undeploying',
  },
];

// 更新状态filter
export const updateStatusFilter = [
  {
    text: '更新成功',
    value: 'success',
  },
  {
    text: '更新失败',
    value: 'fail',
  },
  {
    text: '更新中',
    value: 'update',
  },
];

export const hostsInstallStatus = [
  { text: '管控安装成功', value: '管控安装成功' },
  { text: '管控安装失败', value: '管控安装失败' },
  { text: 'script安装成功', value: 'script安装成功' },
  { text: 'script安装失败', value: 'script安装失败' },
  { text: '主机初始化成功', value: '主机初始化成功' },
  { text: '主机初始化失败', value: '主机初始化失败' },
];

export const kubernetesInstallStatus = [
  { text: '管控安装失败', value: '管控安装失败', status: -1 },
  { text: '管控安装成功', value: '管控安装成功', status: 1 },
  { text: '主机初始化失败', value: '主机初始化失败', status: -3 },
  { text: '主机初始化成功', value: '主机初始化成功', status: 3 },
  { text: 'K8S DOCKER初始化失败', value: 'K8S DOCKER初始化失败', status: -5 },
  { text: 'K8S DOCKER初始化成功', value: 'K8S DOCKER初始化成功', status: 5 },
  { text: 'K8S NODE初始化失败', value: 'K8S NODE初始化失败', status: -6 },
  { text: 'K8S NODE初始化成功', value: 'K8S NODE初始化成功', status: 6 },
  { text: 'K8S NODE部署失败', value: 'K8S NODE部署失败', status: -7 },
  { text: 'K8S NODE部署成功', value: 'K8S NODE部署成功', status: 7 },
];

export const clusterTypeMap = {
  hosts: ['主机集群'],
  kubernetes: ['Kubernetes集群（自建集群）', 'Kubernetes集群（导入已有集群）'],
};

// k8s & hosts 主机初始化状态
export const hostStatusInfoMap = {
  hosts: {
    title:
      '主机安装会经过2个状态：管控安装-主机初始化，若某一步安装失败，请查看具体日志。',
    filters: hostsInstallStatus,
    finalStatus: 3,
  },
  kubernetes: {
    title:
      'Kubernetes集群主机安装会经过5个状态：管控安装-主机初始化-K8S Docker初始化-K8S Node初始化-K8S Node部署成功，若某一步安装失败，请查看具体日志',
    filters: kubernetesInstallStatus,
    finalStatus: 7,
  },
};

export const COOKIES = {
  NAMESPACE: 'em_current_k8s_namespace',
};
