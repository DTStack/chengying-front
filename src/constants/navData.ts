export const navData = [
  {
    title: '运维中心',
    icon: 'emicon-jiqunguanli',
    code: 'menu_ops_center',
    url: '/opscenter',
    level: 'first',
    routers: [
      '/opscenter/overview',
      '/opscenter/service',
      '/opscenter/hoststatus',
      'sub_menu_diagnose',
    ],
    children: [
      {
        title: '概览',
        icon: 'emicon-gailan',
        code: 'sub_menu_overview',
        url: '/opscenter/overview',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '服务',
        icon: 'emicon-fuwu',
        code: 'sub_menu_service',
        url: '/opscenter/service',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '主机',
        icon: 'emicon-zhuji',
        code: 'sub_menu_host',
        url: '/opscenter/hoststatus',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '诊断',
        icon: 'emicon-zhenduan',
        code: 'sub_menu_diagnose',
        url: '/opscenter/diagnosis',
        level: 'second',
        routers: [
          '/opscenter/diagnosis/log',
          '/opscenter/diagnosis/event',
          '/opscenter/diagnosis/config',
          '/opscenter/diagnosis/inspectionReport',
          '/opscenter/diagnosis/backup',
        ],
        children: [
          {
            title: '日志查看',
            icon: 'emicon-rizhichakan',
            code: 'sub_menu_diagnose_log',
            url: '/opscenter/diagnosis/log',
            level: 'three',
            routers: [],
            children: [],
          },
          {
            title: '事件诊断',
            icon: 'emicon-shijianzhenduan',
            code: 'sub_menu_diagnose_event',
            url: '/opscenter/diagnosis/event',
            level: 'three',
            routers: [],
            children: [],
          },
          {
            title: '配置变更',
            icon: 'emicon-peizhibiangeng',
            code: 'sub_menu_diagnose_config_change',
            url: '/opscenter/diagnosis/config',
            level: 'three',
            routers: [],
            children: [],
          },
          {
            title: '巡检报告',
            icon: 'emicon-xunjianbaogao',
            code: 'sub_menu_diagnose',
            url: '/opscenter/diagnosis/inspectionReport',
            level: 'three',
            routers: [],
            children: [],
          },
          {
            title: '备份管理',
            icon: 'emicon-beifenguanli',
            code: 'sub_menu_diagnose_backup',
            url: '/opscenter/diagnosis/backup',
            level: 'three',
            routers: [],
            children: [],
          },
        ],
      },
    ],
  },
  {
    title: '部署中心',
    icon: 'emicon-yingyongguanli',
    code: 'menu_deploy_center',
    url: '/deploycenter',
    level: 'first',
    routers: [
      '/deploycenter/cluster',
      '/deploycenter/appmanage',
      '/deploycenter/deployed',
    ],
    children: [
      {
        title: '集群管理',
        icon: 'emicon-jiqunguanli1',
        code: 'sub_menu_cluster_manage',
        url: '/deploycenter/cluster',
        level: 'second',
        routers: [
          '/deploycenter/cluster/list',
          '/deploycenter/cluster/create',
          '/deploycenter/cluster/create/edit',
          '/deploycenter/cluster/create/add',
          '/deploycenter/cluster/detail/index',
          '/deploycenter/cluster/detail/host',
          '/deploycenter/cluster/detail/imagestore',
          '/deploycenter/cluster/detail/namespace',
          '/deploycenter/cluster/detail/deployed',
          '/deploycenter/cluster/detail/history',
          '/deploycenter/cluster/detail/patchHistory',
          '/deploycenter/cluster/detail/echoList',
          '/deploycenter/cluster/detail/echoDetail',
        ],
        children: [
          {
            title: '集群列表',
            icon: 'emicon-rizhichakan',
            code: 'sub_menu_cluster_list',
            url: '/deploycenter/cluster/list',
            level: 'three',
            routers: [],
            children: [],
          },
          {
            title: '添加集群',
            icon: 'emicon-shijianzhenduan',
            code: 'sub_menu_cluster_edit',
            url: '/deploycenter/cluster/create',
            level: 'three',
            routers: [],
            children: [],
          },
        ],
      },
      {
        title: '组件管理',
        icon: 'emicon-zujianguanli',
        code: 'sub_menu_product_manage',
        url: '/deploycenter/appmanage',
        level: 'second',
        routers: [
          '/deploycenter/appmanage/products',
          '/deploycenter/appmanage/installs',
        ],
        children: [
          {
            title: '组件列表',
            icon: 'emicon-rizhichakan',
            code: 'sub_menu_product_list',
            url: '/deploycenter/appmanage/products',
            level: 'three',
            routers: [],
            children: [],
          },
          {
            title: '部署组件',
            icon: 'emicon-shijianzhenduan',
            code: 'sub_menu_product_deploy',
            url: '/deploycenter/appmanage/installs',
            level: 'three',
            routers: [],
            children: [],
          },
        ],
      },
      {
        title: '告警监控',
        icon: 'emicon-jiankonggaojing2',
        code: 'sub_menu_monitor',
        url: '/deploycenter/deployed',
        level: 'second',
        routers: [
          '/deploycenter/monitoring/dashboard',
          '/deploycenter/monitoring/dashdetail',
          '/deploycenter/monitoring/alert',
          '/deploycenter/monitoring/addAlert',
        ],
        children: [
          {
            title: '仪表盘',
            icon: 'emicon-rizhichakan',
            code: 'sub_menu_dashboard',
            url: '/deploycenter/monitoring/dashboard',
            level: 'three',
            routers: [],
            children: [],
          },
          {
            title: '告警',
            icon: 'emicon-shijianzhenduan',
            code: 'sub_menu_alert',
            url: '/deploycenter/monitoring/alert',
            level: 'three',
            routers: [],
            children: [],
          },
        ],
      },
    ],
  },
  {
    title: '平台管理',
    icon: 'emicon-yingyongguanli',
    code: 'menu_platform_manager',
    url: '/platform',
    level: 'first',
    routers: ['/platform/scriptManager', '/platform/backup', '/platform/clusterInspection'],
    children: [
      {
        title: '备份配置',
        icon: 'emicon-shijianzhenduan',
        code: 'sub_menu_configuration_backup_config',
        url: '/platform/backup',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '脚本管理',
        icon: 'emicon-yunweijiaobenpingtaibeifen',
        code: 'sub_menu_platform_manager_scriptManager',
        url: '/platform/scriptManager',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '集群巡检',
        icon: 'emicon-yunweijiaobenpingtaibeifen',
        code: 'sub_menu_platform_manager_cluster_inspection',
        url: '/platform/clusterInspection',
        level: 'second',
        routers: [],
        children: [],
      },
    ],
  },
  {
    title: '系统配置',
    icon: '',
    code: 'menu_system_configuration',
    url: '/systemconfig',
    level: 'first',
    routers: ['/systemconfig/security', '/systemconfig/globalConfig'],
    children: [
      {
        title: '平台安全',
        icon: 'emicon-jiankonggaojing2',
        code: 'sub_menu_configuration_platformsecurity_config',
        url: '/systemconfig/security',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '全局配置',
        icon: 'emicon-shijianzhenduan',
        code: 'sub_menu_configuration_platformsecurity_config_view',
        url: '/systemconfig/globalConfig',
        level: 'second',
        routers: [],
        children: [],
      },
    ],
  },
  {
    title: '用户管理',
    icon: '',
    code: 'menu_user_manage',
    url: '/usercenter',
    level: 'first',
    routers: [
      '/usercenter/members',
      '/usercenter/role',
      '/usercenter/role/view',
      '/usercenter/selfinfo',
      '/usercenter/audit',
    ],
    children: [
      {
        title: '成员管理',
        icon: 'emicon-chengyuanguanli',
        code: 'sub_menu_user_manage',
        url: '/usercenter/members',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '角色管理',
        icon: 'emicon-jiaoseguanli',
        code: 'sub_menu_role_manage',
        url: '/usercenter/role',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '个人信息',
        icon: 'emicon-gerenxinxi',
        code: 'sub_menu_user_info',
        url: '/usercenter/selfinfo',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '安全审计',
        icon: 'emicon-shenji',
        code: 'menu_security_audit',
        url: '/usercenter/audit',
        level: 'second',
        routers: [],
        children: [],
      },
    ],
  },
];

export const clusterNavData = [
  {
    title: '总览',
    icon: 'emicon-gailan',
    code: 'sub_menu_detail_overview',
    isShow: true,
    key: 'index',
  },
  {
    title: '主机资源',
    icon: 'emicon-fuwu',
    code: 'sub_menu_detail_host',
    isShow: true,
    key: 'host',
  },
  {
    title: '镜像仓库',
    icon: '',
    code: 'sub_menu_detail_image_store',
    isShow: true,
    key: 'imagestore',
  },
  {
    title: '命名空间',
    icon: '',
    code: 'sub_menu_detail_namespace',
    isShow: true,
    key: 'namespace',
  },
  {
    title: '已部署组件',
    icon: '',
    code: 'sub_menu_detail_deployed',
    isShow: true,
    key: 'deployed',
  },
  {
    title: '部署历史',
    icon: '',
    code: 'sub_menu_detail_history',
    isShow: true,
    key: 'history',
  },
  {
    title: '补丁包',
    icon: '',
    code: 'sub_menu_detail_patch',
    isShow: true,
    key: 'patchHistory',
  },
  {
    title: '集群命令',
    icon: '',
    isShow: true,
    key: 'echoList',
  },
  {
    title: '集群命令详情',
    icon: '',
    isShow: false,
    key: 'echoDetail',
  },
];
