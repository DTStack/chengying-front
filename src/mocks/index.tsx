export { default as clusterMnagerMock } from './clusterMnagerMock';
export { default as ServiceMock } from './serviceMock';
export { default as ImageStoreMock, imageStoreInfo } from './imageStoreMock';
export { default as ServicePageMock } from './servicePageMock';
export { default as DeployMock } from './deployMock';
export { default as DashBoardMock } from './dashboardMock';
export { default as UserCenterMock } from './userCenterMock';
export { default as SecurityAuditMock } from './securityAuditMock';
export { default as ClusterNamespaceMock } from './clusterNamespaceMock';
export { cur_parent_cluster } from './clusterMock';
export { authorityList } from './authorityMock';
