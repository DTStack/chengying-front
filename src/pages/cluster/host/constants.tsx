import * as React from 'react';
import { Badge, Tag, Tooltip, Icon, message } from 'antd';
import ProgressBar from '@/components/progressBar';
import './style.scss';

export const columnGenerator = ({
  statusInfo,
  isKubernetes,
  isKubernetesCustom,
  authorityList,
  progressProp,
  hostGroups,
  handleRoleSet,
}) => {
  const roleAuth = authorityList.sub_menu_role_manage;
  return [
    {
      title: '节点',
      dataIndex: 'hostname',
      render: (text, record) => {
        return (
          <div>
            <p>{record.hostname}</p>
            <p className="text-gray">{record.ip}</p>
          </div>
        );
      },
    },
    {
      title: 'agent状态',
      dataIndex: 'is_running',
      filters: [
        { text: '运行中', value: 'true' },
        { text: '已停止', value: 'false' },
      ],
      filterMultiple: true,
      render: (text) => {
        return (
          <Badge
            color={text ? '#12BC6A' : '#FF5F5C'}
            text={text ? '运行中' : '已停止'}
          />
        );
      },
    },
    {
      title: (
        <span>
          主机初始化状态
          <Tooltip title={statusInfo.title}>
            <Icon type="info-circle" />
          </Tooltip>
        </span>
      ),
      dataIndex: 'status',
      width: '15%',
      filters: statusInfo.filters,
      filterMultiple: true,
      render: (text, record: any) => {
        const finalStatus = statusInfo.finalStatus;
        return (
          <React.Fragment>
            <Badge
              color={progressProp(record.status).color}
              text={progressProp(record.status).title || record.errorMsg}
            />
            {record.status !== finalStatus && (
              <Tooltip title={record.errorMsg}>
                &nbsp;
                <Icon type="info-circle" />
              </Tooltip>
            )}
          </React.Fragment>
        );
      },
    },
    isKubernetes
      ? {
          title: '角色',
          dataIndex: 'roles',
          width: '18%',
          filters: [
            { text: 'Control', value: 'Control' },
            { text: 'Worker', value: 'Worker' },
            { text: 'Etcd', value: 'Etcd' },
            { text: '全部', value: 'all' },
          ],
          filterMultiple: true,
          render: (text, record: any) => {
            const roles = [];
            for (const key in record.roles) {
              record.roles[key] && roles.push(key);
            }
            return (
              <React.Fragment>
                {roles.map((item) => (
                  <Tag className="c-cluster__tag" key={item}>
                    {item}
                  </Tag>
                ))}
              </React.Fragment>
            );
          },
        }
      : undefined,
    {
      title: '主机分组',
      dataIndex: 'group',
      filters: hostGroups,
      filterMultiple: true,
    },
    {
      title: 'CPU',
      dataIndex: 'cpu_usage_pct',
      sorter: true,
      render: (text, record: any) => {
        return (
          <ProgressBar
            now={record.cpu_core_used_display}
            total={record.cpu_core_size_display}
          />
        );
      },
    },
    isKubernetes
      ? {
          title: 'POD',
          dataIndex: 'pod_usage_pct',
          sorter: true,
          render: (text, record: any) => {
            return (
              <ProgressBar
                now={record.pod_used_display}
                total={record.pod_size_display}
              />
            );
          },
        }
      : {
          title: '磁盘',
          dataIndex: 'disk_usage_pct',
          sorter: true,
          render: (text, record: any) => {
            return (
              <ProgressBar
                pStyle={{ width: 140, marginLeft: '-28px' }}
                now={record.disk_used_display}
                total={record.disk_size_display}
              />
            );
          },
        },
    !isKubernetesCustom
      ? {
          title: '角色',
          dataIndex: 'role_list_display',
          ellipsis: true,
          render: (roleList = []) => {
            return roleList.length === 0 ? (
              '--'
            ) : (
              <Tooltip title={roleList.map((role) => role.role_name).join(',')}>
                <span>{roleList.map((role) => role.role_name).join(',')}</span>
              </Tooltip>
            );
          },
        }
      : undefined,
    !isKubernetesCustom
      ? {
          title: '操作',
          render: (value, record) => {
            record.role_list_display = record.role_list_display || [];
            const roleList = record.role_list_display.map(
              (item) => item.role_id
            );
            return (
              <span
                className="link-btn"
                onClick={() => {
                  if (roleAuth) {
                    handleRoleSet(record.sid, roleList);
                  } else {
                    message.error('权限不足，请联系管理员！');
                  }
                }}>
                编辑角色
              </span>
            );
          },
        }
      : undefined,
  ].filter((item) => item !== undefined);
};
