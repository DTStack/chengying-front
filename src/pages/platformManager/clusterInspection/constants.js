export const form_Title_info = [
    {
        "form_title": 'service_gc_statistics',
        "desc": "服务 Full GC 统计"
    },
    {
        "form_title": 'dir_usage_statistics',
        "desc": "/opt/dtstack/目录使用统计"
    },
    {
        "form_title": 'alarm_indicators',
        "desc": "告警指标列表"
    }
]

export const chartTable = [
    {
        'insert': 'HDFS 文件存储使用率',
        "form_title": 'hdfs_file_usage',
        "desc": "HDFS 文件存储使用率列表"
    },
    {
        'insert': 'CPU使用率趋势',
        "form_title": 'cpu_usage',
        "desc": "CPU使用率列表"
    },
    {
        'insert': '内存使用率趋势',
        "form_title": 'men_usage',
        "desc": "内存使用率列表 (最新)"
    },
    {
        'insert': 'swap 使用量趋势',
        "form_title": 'swap_usage',
        "desc": "swap使用列表 (最新)"
    },
    {
        'insert': '磁盘使用率趋势',
        "form_title": 'disk_usage',
        "desc": "磁盘使用率列表 (最新)"
    },
    {
        'insert': 'inode 使用率趋势',
        "form_title": 'inode_usage',
        "desc": "inode 使用率列表 (最新)"
    },
]
export const colorRange = [
    '#2FC25B',
    '#1890FF',
    '#6087EF',
    '#7460EF',
    '#AB60EF',
    '#E85EB3',
    '#EF5350',
    '#F18646',
    '#FFB310',
    '#8BDA63',
  ];

export const bigDataNodeList = [
    {
        "bigData_Node_title": 'NameNode内存(最新)',
        "name_node_memo": ""
    },
    {
        "bigData_Node_title": 'DataNode内存(最新)',
        "data_node_mem": ""
    },
    {
        "bigData_Node_title": 'DataNode存活数(最新)',
        "data_node_live_nums": ""
    },
    {
        "bigData_Node_title": 'DataNode死亡数(最新)',
        "data_node_dead_nums": ""
    },
    {
        "bigData_Node_title": 'HDFS文件数(最新)',
        "hdfs_file_num": ""
    }
]

