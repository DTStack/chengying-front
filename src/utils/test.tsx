import * as React from 'react';
import { render } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import thunkMiddleware from 'redux-thunk';

// this is a handy function that I normally make available for all my tests
// that deal with connected components.
// you can provide initialState for the entire store that the ui is rendered with
export function renderWithRedux(
  Component,
  rootReducer,
  initialState: any = {}
) {
  const store: any = createStore(
    rootReducer,
    initialState,
    applyMiddleware(thunkMiddleware)
  );
  const layout = <Provider store={store}>{Component}</Provider>;
  return {
    ...render(layout),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  };
}
