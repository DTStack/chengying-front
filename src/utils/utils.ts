import { message } from 'antd';
import * as Cookies from 'js-cookie';
import { COOKIES } from '@/constants/const';

// tslint:disable
const utils = {
  // 公用的方法
  // loginOut: function () {
  //     Http.get(req_url.login_out,{}).then(ret => {
  //         Http.get(req_url.get_config,{}).then(data => {
  //             window.location.href = uicUrl + ":" + uicPort + redirectLink + window.location.href;
  //         })
  //         return ret;
  //     })
  // },
  /**
   * 获取页面宽度
   * @return {[type]} [description]
   */
  pageWidth: function () {
    return Math.max(
      document.documentElement.clientWidth,
      window.innerWidth || 0
    );
  },

  /**
   * 获取页面高度
   * @return {[type]} [description]
   */
  pageHeight: function () {
    return Math.max(
      document.documentElement.clientHeight,
      window.innerHeight || 0
    );
  },

  /**
   * 根据参数名获取URL数据
   * @param  {[type]} name [description]
   * @param  {[type]} url  [description]
   * @return {[type]}      [description]
   */
  getParameterByName: function (name: any, url: any) {
    if (!url) {
      url = window.location.href;
    }
    name = name.replace(/[[\]]/g, '\\$&');
    var regex = new RegExp('[?&]' + name + '(=([^&#]*)|&|#|$)');
    var results = regex.exec(url);
    if (!results) {
      return null;
    }
    if (!results[2]) {
      return '';
    }
    return decodeURIComponent(results[2].replace(/\+/g, ' '));
  },

  /**
   * 获取图片的Base64格式
   * @param  {[type]}   img      [description]
   * @param  {Function} callback [description]
   * @return {[type]}            [description]
   */
  getBase64: function (img: any, callback: any) {
    const reader = new FileReader();
    reader.addEventListener('load', () => callback(reader.result));
    reader.readAsDataURL(img);
  },

  /**
   * 百分比转换
   * @param  {[type]} num       [description]
   * @param  {[type]} precision [description]
   * @return {[type]}           [description]
   */
  percent: function (num: any, precision: any) {
    if (!num || num === Infinity) {
      return 0 + '%';
    }
    if (num > 1) {
      num = 1;
    }
    precision = precision || 2;
    precision = Math.pow(10, precision);
    return Math.round(num * precision * 100) / precision + '%';
  },

  getCssText: function (object: any) {
    var str = '';
    for (var attr in object) {
      str += attr + ':' + object[attr] + ';';
    }
    return str;
  },

  formateDateTime: function (timestap: any) {
    if (!timestap) {
      return '';
    }
    const times = new Date(timestap);
    const year = times.getFullYear();
    let month = times.getMonth() + 1 + '';
    let day = times.getDate() + '';

    let hour = times.getHours() + '';
    let minutes = times.getMinutes() + '';
    let seconds = times.getSeconds() + '';

    if (month.toString().length === 1) {
      month = `0${month}`;
    }

    if (day.toString().length === 1) {
      day = `0${day}`;
    }

    if (hour.toString().length === 1) {
      hour = `0${hour}`;
    }

    if (minutes.toString().length === 1) {
      minutes = `0${minutes}`;
    }

    if (seconds.toString().length === 1) {
      seconds = `0${seconds}`;
    }
    return `${year}-${month}-${day} ${hour}:${minutes}:${seconds}`;
  },

  formateDate: function (timestap: any) {
    if (!timestap) {
      return '';
    }
    const times = new Date(timestap);
    const year = times.getFullYear();
    let month = times.getMonth() + 1 + '';
    let day = times.getDate() + '';
    if (month.toString().length === 1) {
      month = `0${month}`;
    }
    if (day.toString().length === 1) {
      day = `0${day}`;
    }
    return `${year}-${month}-${day}`;
  },

  trim: function (str: any) {
    return typeof str === 'string'
      ? str.replace(/^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g, '')
      : str;
  },

  messageFilter: function (
    data: any,
    callback: any,
    info?: any,
    hideSuccessMessage?: any
  ) {
    if (data.status.code === 1001) {
      // if(info){
      //     message.success(info);
      // }
      // message.success(data.status.msg);
      if (!hideSuccessMessage) {
        message.success('成功');
      }
      if (callback) {
        callback();
      }
    } else {
      // console.log(data.status.msg)
      // message.error(data.status.msg);
    }
  },

  // filter查询条件的构造
  getTableParam: (filter: any) => {
    const param = {};
    filter.forEach((item: any, i: any) => {
      const keyArr = item.split('-');
      param[keyArr[1]] = keyArr[2];
    });
    return param;
  },

  // 抑制告警
  setAlertControl: (data: any) => {
    const alarmSuppression: any = {};
    const acKey = data.alert_control;
    if (acKey === '1') {
      alarmSuppression.close = 1;
    } else {
      alarmSuppression.close = 0;
      if (data['alert_num_' + acKey]) {
        alarmSuppression.times_after_close = data['alert_num_' + acKey];
      }
      if (data['recover_num_' + acKey]) {
        alarmSuppression.close_after_long_time_reset_amount =
          data['recover_num_' + acKey];
        alarmSuppression.close_after_long_time_reset_unit =
          data['recover_unit_' + acKey];
      }
    }
    return alarmSuppression;
  },
  getAlertControl: (alertControl: any) => {
    const fields = {
      alert_control: '',
    };
    if (alertControl.close === 0) {
      if (alertControl.times_after_close !== 0) {
        fields.alert_control = '2';
        fields['alert_num_' + fields.alert_control] =
          alertControl.times_after_close;
      } else {
        fields.alert_control = '3';
      }
      if (alertControl.close_after_long_time_reset_amount !== '') {
        if (fields.alert_control === '2') {
          fields.alert_control = '4';
        }
        fields['alert_num_' + fields.alert_control] =
          alertControl.times_after_close;
        fields['recover_num_' + fields.alert_control] =
          alertControl.close_after_long_time_reset_amount;
        fields['recover_unit_' + fields.alert_control] =
          alertControl.close_after_long_time_reset_unit;
      }
    }
    return fields;
  },

  // 模糊匹配
  filterOption: (inputValue: any, option: any) => {
    if (option.props.children.indexOf(inputValue) !== -1) {
      return true;
    } else {
      return false;
    }
  },
  // 从url中分离参数
  getParamsFromUrl(url: any) {
    var obj = {};
    var keyvalue = [];
    var key = '';
    var value = '';
    var paraString = url.substring(url.indexOf('?') + 1, url.length).split('&');
    for (var i in paraString) {
      keyvalue = paraString[i].split('=');
      key = keyvalue[0];
      value = keyvalue[1];
      obj[key] = value;
    }
    return obj;
  },
  jsonToQuery(obj: any) {
    function cleanArray(actual: any) {
      const newArray = [];
      for (let i = 0; i < actual.length; i++) {
        if (actual[i]) {
          newArray.push(actual[i]);
        }
      }
      return newArray;
    }
    if (!obj) {
      return '';
    }
    return cleanArray(
      Object.keys(obj).map((key) => {
        if (obj[key] === undefined) {
          return '';
        }
        return encodeURIComponent(key) + '=' + encodeURIComponent(obj[key]);
      })
    ).join('&');
  },
  // 判断Object是否为{}
  checkNullObj(obj: any) {
    return Object.keys(obj).length === 0;
  },

  // service中对于简单res的处理
  serviceCallback(res: any, successMsg?: string, callback?: Function) {
    if (res.data.code === 0) {
      message.success(successMsg);
      callback && callback();
    } else {
      message.error(res.data.msg);
    }
  },

  // 是否有权限操作
  noAuthorityToDO(authorityList: any, type: string) {
    if (!authorityList[type]) {
      message.error('权限不足，请联系管理员!');
      return true;
    }
    return false;
  },

  get k8sNamespace(): string {
    return Cookies.get(COOKIES.NAMESPACE);
  },

  setNaviKey: (top: string | undefined, sub: string | undefined): void => {
    sessionStorage.setItem('firstLevelNav', top);
    sessionStorage.setItem('siderLevelNav', sub);
  },

  // 转化为GB
  formatGBUnit: (r: string): number => {
    let result: number = 0;
    if (r.indexOf('MB') > -1) {
      result = parseFloat(r.replace('MB', '')) / 1024;
    } else if (r.indexOf('KB') > -1) {
      result = parseFloat(r.replace('KB', '')) / 1024 / 1024;
    } else if (r.indexOf('GB') > -1) {
      result = parseFloat(r.replace('GB', ''));
    } else if (r.indexOf('TB') > -1) {
      result = parseFloat(r.replace('TB', '')) * 1024;
    } else {
      result = 0;
    }
    return result;
  },
};

export default utils;
