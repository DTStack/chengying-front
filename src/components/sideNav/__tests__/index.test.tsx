import * as React from 'react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import SideNav from '../index';
import { renderWithRedux } from '@/utils/test';
import { authorityList } from '@/mocks';
import reducer from '@/stores';
import { navData } from '@/constants/navData';
import {
  fireEvent,
  RenderResult,
  cleanup,
  screen,
} from '@testing-library/react';

const defaultProps = {
  selectKeys: ['sub_menu_diagnose_log'],
  menuList: navData[0].children,
  defaultOpenKey: ['sub_menu_diagnose'],
  authorityList,
  history: {},
};

describe('sidenav render', () => {
  let wrapper: RenderResult;
  beforeEach(() => {
    const history = createMemoryHistory();
    const layout = (
      <Router history={history}>
        <SideNav {...defaultProps} />
      </Router>
    );
    wrapper = renderWithRedux(layout, reducer, {
      UserCenterStore: { authorityList },
    });
  });

  afterEach(cleanup);

  // menuList
  test('menuList render', () => {
    const menuItems = wrapper.container.getElementsByClassName('ant-menu-item');
    const menuItemNames = ['概览', '服务', '主机'];
    for (const i in menuItems) {
      if (menuItems[i] instanceof HTMLElement) {
        expect(menuItems[i].textContent).toBe(menuItemNames[i]);
      }
    }
  });

  // submenu
  // test('submenu render', () => {
  //     screen.debug();
  //     const subMenu = wrapper.container.getElementsByClassName('ant-menu-sub');
  //     expect(subMenu.container.getElementsByClassName('ant-menu-item')[0].textContent).toBe('日志查看');
  //     // defaultOpenKey
  //     const subMenuLi = subMenu.parentElement;
  //     expect(subMenuLi.className).toMatch('ant-menu-submenu-open');
  //     // 收起
  //     fireEvent.click(subMenu);
  //     expect(subMenuLi.className).not.toMatch('ant-menu-submenu-open');
  // })

  test('sider snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
