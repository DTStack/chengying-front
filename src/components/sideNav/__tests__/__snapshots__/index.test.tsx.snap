// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidenav render sider snapshot 1`] = `
<DocumentFragment>
  <div
    class="dt-sidemenu-dark"
    style="min-height: calc(100vh - 40px); height: 100%;"
  >
    <ul
      class="ant-menu ant-menu-dark ant-menu-root ant-menu-inline"
      role="menu"
      style="width: 200px;"
    />
  </div>
</DocumentFragment>
`;
